File: app/src/main/java/com/geridea/trentastico/gui/adapters/YearsAdapter.java
package com.geridea.trentastico.gui.adapters;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.alexvasilkov.android.commons.adapters.ItemsAdapter;

import java.util.ArrayList;

import com.geridea.trentastico.R;

public class YearsAdapter extends ItemsAdapter<Integer> {


    public YearsAdapter(Context context) {
        super(context);

        ArrayList<Integer> integers = new ArrayList<>();
        integers.add(1);
        integers.add(2);
        integers.add(3);
        integers.add(4);
        integers.add(5);

        setItemsList(integers);
    }

    @Override
    protected View createView(Integer item, int pos, ViewGroup parent, LayoutInflater inflater) {
        return inflater.inflate(R.layout.itm_spinner, parent, false);
    }

    @Override
    protected void bindView(Integer item, int pos, View textView) {
        ((TextView) textView.findViewById(R.id.text)).setText(item.toString());
    }

    @Override
    public long getItemId(int pos) {
        return getItem(pos);
    }
}

File: app/src/main/java/com/geridea/trentastico/gui/adapters/YearsAdapter.kt
package com.geridea.trentastico.gui.adapters


import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.alexvasilkov.android.commons.adapters.ItemsAdapter
import com.geridea.trentastico.R
import com.geridea.trentastico.model.StudyYear

class YearsAdapter(context: Context, years: List<StudyYear>) : ItemsAdapter<StudyYear>(context) {

    init {
        itemsList = years
    }

    override fun createView(
            item: StudyYear?,
            pos: Int,
            parent: ViewGroup,
            inflater: LayoutInflater): View = inflater.inflate(R.layout.itm_spinner, parent, false)

    override fun bindView(item: StudyYear, pos: Int, view: View) {
        (view.findViewById(R.id.text) as TextView).text = item.name
    }

    override fun getItemId(pos: Int): Long = pos.toLong()

    fun getPositionOfYearWithId(idToSearch: String): Int? =
            (0 until itemsList.size).firstOrNull { itemsList[it].id == idToSearch }
}
